mashumaro/core/meta/helpers.py

```diff
@@ -289,7 +286,7 @@ def type_name(
 def is_special_typing_primitive(typ: Any) -> bool:
     try:
         issubclass(typ, object)
-        return False
+        return PY_314_MIN and issubclass(typ, typing.Union)  # type: ignore[arg-type]
```

Comment:
What case do we need this for? I mean, when will this be `True`? On Python 3.14.0rc3 all the following statements seem to throw `TypeError`:
* `issubclass(int | str, object)`
* `issubclass(Union[int, str], object)`
* `issubclass(UnionType[int, str], object)`
* `issubclass(int | str, typing.Union)`
* `issubclass(Union[int, str], typing.Union)`
* `issubclass(UnionType[int, str], typing.Union)`